source_gen/lib/src/builder.dart

```diff
@@ -69,7 +70,14 @@ class _Builder extends Builder {
   @override
   Future build(BuildStep buildStep) async {
     final resolver = buildStep.resolver;
+
     if (!await resolver.isLibrary(buildStep.inputId)) return;
+
+    if (_generators.every((g) => g is GeneratorForAnnotation) &&
+        !await _hasAnyTopLevelAnnotations(buildStep.inputId, resolver)) {
```

Comment:
[nit] I find this a bit easier to read if it has parens
```suggestion
        !(await _hasAnyTopLevelAnnotations(buildStep.inputId, resolver))) {
```